sklearn/cluster/_agglomerative.py

```diff
@@ -984,8 +1017,7 @@ def fit(self, X, y=None):
         return self
 
     def fit_predict(self, X, y=None):
-        """Fit the hierarchical clustering from features or distance matrix,
-        and return cluster labels.
+        """Fit the hierarchical clustering from features, or distance matrix.
 
```

Comment:
Please add a short paragraph to make the difference method between the `fit_predict` and the `fit` methods explicit,  for instance: "In addition to fitting, this method also return the result of the clustering assignment for each sample in the training set."